app/src/scripts/common/forms/paramTypes/paramController.jsx

```diff
@@ -0,0 +1,280 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Input from '../input.jsx'
+
+class ParamController extends React.Component {
```

Comment:
This should be a parent component and a type specific one since most of the logic is 'if (type == something) { do something behavior }'

Reply:
I have started breaking up all of these input types. It will take me a day or two to get full functionality back up and running.

Reply:
It is on a separate branch called: input-update-refactor. Will merge it into this branch for final review?